src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPools.cs

```diff
@@ -66,6 +77,12 @@ private void RemoveStalePools()
                 }
             }
 
+            // Stop running the timer if we don't have any pools to clean up.
+            if (_pools.IsEmpty)
+            {
+                _cleaningTimer.Change(Timeout.Infinite, Timeout.Infinite);
```

Comment:
There's a race condition here.  Imagine there's one connection in the pool and it's stale.  RemoveStalePools runs and removes it, then checks and sees that pools is empty, so it goes to stop the timer.  At that same moment, another thread comes along, sees that the pool is empty, adds a connection, and changes the timer to fire.  Now this thread continues running and stops the timer.  We now then have connections but no timer running.